src/rgw/support/rgw_support_bundle

```diff
@@ -117,32 +133,95 @@ class RGWSupportBundle:
+            dest_dir_size = RGWSupportBundle._get_dir_size(
+                RGWSupportBundle._tmp_src)
+            client_log_dir_size = RGWSupportBundle._get_dir_size(tmp_client_log_dir)
+            Log.info(f'Total size of textual log files: {client_log_dir_size}')
+            remaining_size_limit = size_limit_in_byte - dest_dir_size
+            Log.info(f'Remaining size_limit for textual log : {remaining_size_limit}')
+            if (remaining_size_limit >= client_log_dir_size):
+                for file_regex in ['rgw*', 'radosgw*']:
+                    RGWSupportBundle._apply_size_filter(
+                        tmp_client_log_dir, size_limit, file_regex)
+            else:
+                tmp_svc_files_dir = os.path.join('/tmp', 'rgw_svc_log')
+                os.makedirs(tmp_svc_files_dir, exist_ok=True)
+                for file in os.listdir(tmp_client_log_dir):
+                    if file.startswith('rgw-') and file.endswith('.log'):
+                        shutil.move(os.path.join(tmp_client_log_dir, file),
+                            os.path.join(tmp_svc_files_dir, file))
+                # Allocate 95% of size from total size_limit to
+                # rgw svc log(rgw-1.log) files
+                allocated_size = str(((
+                    size_limit_in_byte * SVC_FILE_PERCENTAGE)/100)) + 'B'
+                Log.info(f'Allocated size for {COMPONENT_NAME} svc files:{allocated_size}')
+                RGWSupportBundle._apply_size_filter(
+                    tmp_svc_files_dir, allocated_size, 'rgw-*',
+                    check_remaining_size=False)
+                RGWSupportBundle._copy_compressed_files(tmp_client_log_dir,
+                    allocated_size)
+                RGWSupportBundle._apply_size_filter(
+                    tmp_client_log_dir, size_limit, 'rgw_*')
+                RGWSupportBundle._apply_size_filter(
+                    tmp_client_log_dir, size_limit, 'radosgw*')
+                shutil.rmtree(tmp_svc_files_dir)
+            shutil.rmtree(tmp_client_log_dir)
 
         # copy motr trace log files
         motr_trace_dir = os.path.join(log_dir, 'motr_trace_files')
         if os.path.exists(motr_trace_dir):
+            tmp_motr_trace_dir = os.path.join(RGWSupportBundle._tmp_src, 'motr_trace_files')
             # include the latest 5 log files of motr traces in support bundle
-            list_of_files = filter(lambda f: os.path.isfile(os.path.join(motr_trace_dir, f)),
-                            os.listdir(motr_trace_dir))
-            # sort the files based on last modification time
-            list_of_files = sorted(list_of_files,
-                            key = lambda f: os.path.getmtime(os.path.join(motr_trace_dir, f)),
-                            reverse=True)
-            list_of_files = list_of_files[0:5]
-            for file in list_of_files:
+            latest_files = RGWSupportBundle._get_latest_files(directory=motr_trace_dir)
```

Comment:
If size_limit filter is not applied we are collecting the latest 5 motr trace files,
otherwise, if motr_trace files fit in the given size_limit then only we are including those files in the support bundle.

@yeshpal-jain-seagate , @nkommuri Please confirm if this is fine.

Reply:
It is fine.